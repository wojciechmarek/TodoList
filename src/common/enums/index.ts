export * from './theme';
export * from './task-status';
