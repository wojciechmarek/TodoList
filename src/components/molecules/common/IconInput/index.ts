export * from './icon-input';
export * from './icon-input.interface';
