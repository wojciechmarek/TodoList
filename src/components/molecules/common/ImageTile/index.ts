export * from './image-tile';
export * from './image-tile.interface';
