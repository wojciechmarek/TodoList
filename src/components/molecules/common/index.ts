export * from './IconButton';
export * from './IconInput';
export * from './ImageTile';
