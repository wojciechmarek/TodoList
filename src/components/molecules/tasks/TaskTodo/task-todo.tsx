import { TaskTodoProps } from './task-todo.interface';
import {
  TaskItemActions,
  TaskItemContainer,
  TaskItemDescription,
  TaskItemDoneIconButton,
  TaskItemRemoveIconButton,
  TaskItemText,
  TaskItemTitleDescription,
} from './task-todo.styled';

import { Check, Trash2 } from 'lucide-react';

import { useCallback } from 'react';

export const TaskTodo = (props: TaskTodoProps) => {
  const { task, handleDoneClick, handleOnItemClick, handleRemoveClick } = props;

  const onContainerClick = useCallback(
    (e: React.MouseEvent<HTMLDivElement>) => {
      e.stopPropagation();
      handleOnItemClick(task.id || 0);
    },
    []
  );

  const onDoneButtonClick = useCallback(
    (e: React.MouseEvent<HTMLButtonElement>) => {
      e.stopPropagation();
      handleDoneClick(task.id || 0);
    },
    []
  );

  const onRemoveButtonClick = useCallback(
    (e: React.MouseEvent<HTMLButtonElement>) => {
      e.stopPropagation();
      handleRemoveClick(task.id || 0);
    },
    []
  );

  return (
    <TaskItemContainer onClick={onContainerClick}>
      <TaskItemTitleDescription>
        <TaskItemText text={props.task.title} />
        <TaskItemDescription text={props.task.description} />
      </TaskItemTitleDescription>
      <TaskItemActions>
        <TaskItemDoneIconButton onClick={onDoneButtonClick} icon={<Check />} />
        <TaskItemRemoveIconButton
          onClick={onRemoveButtonClick}
          icon={<Trash2 />}
        />
      </TaskItemActions>
    </TaskItemContainer>
  );
};
