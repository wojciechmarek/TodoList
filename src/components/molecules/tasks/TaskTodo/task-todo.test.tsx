import { configure, render } from '@testing-library/react';
import { TaskTodoProps } from './task-todo.interface';
import { TaskTodo } from './task-todo';
import '@testing-library/jest-dom';
import { TaskStatus } from '../../../../common';

describe('TaskTodo', () => {
  let taskTodoProps: TaskTodoProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    taskTodoProps = {
      handleOnItemClick: jest.fn(),
      handleRemoveClick: jest.fn(),
      handleDoneClick: jest.fn(),

      task: {
        id: 1,
        title: 'title',
        description: 'description',
        status: TaskStatus.Done,
      },
    };
  });

  it('should render', () => {
    const { baseElement } = render(<TaskTodo {...taskTodoProps} />);
    expect(baseElement).toBeTruthy();
  });

  describe('when the task completed is rendered', () => {
    it('should render the title', () => {
      const { getByText } = render(<TaskTodo {...taskTodoProps} />);
      const title = getByText(/title/i);
      expect(title).toBeTruthy();
    });

    it('should render the description', () => {
      const { getByText } = render(<TaskTodo {...taskTodoProps} />);
      const description = getByText(/description/i);
      expect(description).toBeTruthy();
    });
  });

  describe('when the task completed is clicked', () => {
    it('should call the handleOnItemClick function', () => {
      const { getByText } = render(<TaskTodo {...taskTodoProps} />);
      const task = getByText(/title/i);
      task.click();
      expect(taskTodoProps.handleOnItemClick).toHaveBeenCalled();
    });
  });

  describe('when the remove button is clicked', () => {
    it('should call the handleRestoreClick function', async () => {
      const { getAllByRole } = render(<TaskTodo {...taskTodoProps} />);
      const restoreButton = getAllByRole('button', { hidden: true })[0];
      restoreButton.click();
      expect(taskTodoProps.handleDoneClick).toHaveBeenCalled();
    });
  });

  describe('when the done button is clicked', () => {
    it('should call the handleRemoveClick function', async () => {
      const { getAllByRole } = render(<TaskTodo {...taskTodoProps} />);
      const removeButton = getAllByRole('button', { hidden: true })[1];
      removeButton.click();
      expect(taskTodoProps.handleRemoveClick).toHaveBeenCalled();
    });
  });
});
