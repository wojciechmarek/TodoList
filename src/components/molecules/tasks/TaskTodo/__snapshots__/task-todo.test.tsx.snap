// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTodo should match the snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-height: 3em;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  background-color: var(--task-item-row-bg-color);
  width: 100%;
  margin-bottom: 0.5em;
  border: var(--standard-border);
  border-radius: var(--standard-border-radius);
  box-shadow: var(--button-box-shadow);
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  cursor: pointer;
}

.emotion-0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.emotion-0:hover div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  items-align: center;
}

.emotion-2 {
  margin-left: 1em;
  font-weight: 800;
  color: var(--button-font-color);
}

.emotion-3 {
  margin-left: 1em;
  font-weight: 800;
  -webkit-filter: opacity(0.6);
  filter: opacity(0.6);
  color: var(--button-font-color);
  text-overflow: ellipsis;
}

.emotion-4 {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-right: 0.35em;
  gap: 0.5em;
}

.emotion-5 {
  cursor: pointer;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: none;
  height: 2.25em;
  width: 2.25em;
  border-radius: 0.6em;
  border: var(--standard-border);
  border-radius: var(--standard-border-radius);
  box-shadow: var(--button-box-shadow);
  color: var(--text-font-color);
  background-color: var(--task-item-done-button-bg-color);
}

.emotion-5:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  width: 100%;
}

.emotion-7 {
  height: calc(100% * 0.618);
  width: calc(100% * 0.618);
}

.emotion-8 {
  cursor: pointer;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border: none;
  height: 2.25em;
  width: 2.25em;
  border-radius: 0.6em;
  border: var(--standard-border);
  border-radius: var(--standard-border-radius);
  box-shadow: var(--button-box-shadow);
  color: var(--text-font-color);
  background-color: var(--task-item-delete-button-bg-color);
}

.emotion-8:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

<div
  className="emotion-0"
  onClick={[Function]}
>
  <div
    className="emotion-1"
  >
    <p
      className="emotion-2"
    >
      title
    </p>
    <p
      className="emotion-3"
    >
      description
    </p>
  </div>
  <div
    className="emotion-4"
  >
    <button
      className="emotion-5"
      onClick={[Function]}
    >
      <div
        className="emotion-6"
      >
        <div
          className="emotion-7"
        >
          <svg
            className="lucide lucide-check"
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="20 6 9 17 4 12"
            />
          </svg>
        </div>
      </div>
    </button>
    <button
      className="emotion-8"
      onClick={[Function]}
    >
      <div
        className="emotion-6"
      >
        <div
          className="emotion-7"
        >
          <svg
            className="lucide lucide-trash2"
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 6h18"
            />
            <path
              d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
            />
            <path
              d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
            />
            <line
              x1="10"
              x2="10"
              y1="11"
              y2="17"
            />
            <line
              x1="14"
              x2="14"
              y1="11"
              y2="17"
            />
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;
